lighthouse-core/audits/seo/is-crawlable.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const BLOCKLIST = new Set([
+  'noindex',
+  'none',
+]);
+const ROBOTS_HEADER = 'x-robots-tag';
+const UNAVAILABLE_AFTER = 'unavailable_after';
+
+/**
+ * Checks if given directive is a valid unavailable_after directive with a date in the past
+ * @param {string} directive
+ * @returns {boolean}
+ */
+function isUnavailable(directive) {
+  const parts = directive.split(':');
+
+  if (parts.length <= 1 || parts[0] !== UNAVAILABLE_AFTER) {
+    return false;
+  }
+
+  const date = Date.parse(parts.slice(1).join(':'));
+
+  return !isNaN(date) && date < Date.now();
+}
+
+/**
+ * Returns true if any of provided directives blocks page from being indexed
+ * @param {string} directives
+ * @returns {boolean}
+ */
+function hasBlockingDirective(directives) {
+  return directives.split(',')
+    .map(d => d.toLowerCase().trim())
+    .some(d => BLOCKLIST.has(d) || isUnavailable(d));
+}
+
+/**
+ * Returns true if robots header specifies user agent (e.g. `googlebot: noindex`)
+ * @param {string} directives
+ * @returns {boolean}
+ */
+function hasUserAgent(directives) {
+  const parts = directives.match(/^([^,:]+):/);
+
+  // Check if directives are prefixed with `googlebot:`, `googlebot-news:`, `otherbot:`, etc.
+  // but ignore `unavailable_after:` which is a valid directive
+  return !!parts && parts[1].toLowerCase() !== UNAVAILABLE_AFTER;
+}
+
+class IsCrawlable extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'is-crawlable',
+      description: 'Page isn’t blocked from indexing',
```

Comment:
I'd prefer to word this more affirmatively (i.e. `Page can be indexed` or `Page is indexable`), but I'm guessing we can't because it'd be misleading and there are many other ways a page could be prevented from indexing?